compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -1857,6 +1857,29 @@ LogicalResult TensorCloneOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// flow.tensor.encode
+//===----------------------------------------------------------------------===//
+
+LogicalResult TensorEncodeOp::verify() {
+  if (failed(verifyOpDynamicDims(getOperation(), {getOperand()},
+                                 getOperandDims())) ||
+      failed(verifyOpDynamicDims(getOperation(), {getResult()},
+                                 getOperandDims()))) {
+    return failure();
+  }
+  auto operandType = cast<RankedTensorType>(getOperand().getType());
+  if (operandType.getEncoding()) {
+    return emitOpError("the source operand type has encoding; not allowed");
```

Comment:
we want this to work both directions like a cast so I'd not expect special casing on whether the source or result has encodings - dense -> encoding A, encoding A -> dense, encoding A -> encoding A + dense -> dense (fold), and encoding A -> encoding B (reencode)

Reply:
I see, so I only need to verify shape compatible, or if they are exactly the same type without encodings. Do we allow encoding `tensor<4x?xf32>` to `tensor<?x5xf32, #encoding>`? It could make op definition a little harder because we'll need to carry the result dynamic dimensions as well.